src/Kernel/PyriteKernel.php

```diff
@@ -117,43 +113,95 @@ private function configPhpErrors(\Pyrite\Exception\UncaughtExceptionRenderer $re
             $this->errorHandler = new \Pyrite\Exception\ErrorHandlerImpl(0, false);
             $this->errorHandler->setOnFatalRenderer($renderer);
         }
-
         $this->errorHandler->enable();
 
         return $this->errorHandler;
     }
 
-    private function configOnCrashOutput()
+    /**
+     * {@inheritDoc}
+     */
+    public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true)
     {
+        $context = new RequestContext();
+        $context->fromRequest($request);
+        $urlMatcher = new UrlMatcher($this->routeCollection, $context);
+        $dispatch = null;
         try {
-            $this->uncaughtRenderer = $this->container->get("AppOnCrashHandler");
-        } catch (\Exception $e) {
-            $this->uncaughtRenderer = new \Pyrite\Exception\UncaughtExceptionRendererImpl(false);
+            $parameters = $urlMatcher->match($request->getPathInfo());
+            $dispatch = $this->getDispatchStackFromRoute($parameters);
+            //@TODO improve request bindings
+            $request->attributes->replace($parameters);
+        } catch (ResourceNotFoundException $e) {
+            if ($this->container->getParameter('404')) {
```

Comment:
Ouch ! So here you assume that the route dispatch stack is in the container and not in the router... looks weird :-/